docs/shared/recipes/plugins/create-install-package.md

```diff
@@ -0,0 +1,159 @@
+
+## Generating a "Create Package"
+
+There are a few methods to create a package that will work with `create-nx-workspace`'s public API to setup a new workspace that uses your Nx plugin.
+
+You can setup a new Nx plugin workspace and immediately pass the `--create-package-name`:
+
+```shell
+npx create-nx-plugin my-plugin --create-package-name create-my-plugin
+```
+
+Alternatively, if you already have an existing Nx plugin workspace, you can run the following generator to set up a new create package:
+
+```shell
+nx g create-package create-my-plugin --project my-plugin
+```
+
+## Customize your create package
+
+You'll have 2 packages that are relevant in your workspace:
+
+- The create package (e.g. `create-my-plugin`)
+- The plugin package (e.g. `my-plugin`)
+
+Let's take a look at the code that was scaffolded out for your create-my-plugin package:
+
+```typescript {% fileName="packages/create-my-plugin/bin/index.ts" %}
+#!/usr/bin/env node
+
+import { createWorkspace } from 'create-nx-workspace';
+
+async function main() {
+  const name = process.argv[2]; // TODO: use libraries like yargs or enquirer to set your workspace name
+  if (!name) {
+    throw new Error('Please provide a name for the workspace');
+  }
+
+  console.log(`Creating the workspace: ${name}`);
+
+  // This assumes "my-plugin" and "create-my-plugin" are at the same version
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const presetVersion = require('../package.json').version;
+
+  // TODO: update below to customize the workspace
+  const { directory } = await createWorkspace(`my-plugin@${presetVersion}`, {
+    name,
+    nxCloud: false,
+    packageManager: 'npm',
+  });
+
+  console.log(`Successfully created the workspace: ${directory}.`);
+}
+
+main();
+```
+
+This is a plain node script at this point, and you can use any dependencies you wish to handle things like prompting or argument parsing. Keeping dependencies small and splitting out the command line tool from the Nx plugin is recommended, and will help keep your CLI feel fast and snappy.
+
+Note the following code snippet:
+
+```typescript
+const { directory } = await createWorkspace(`my-plugin@${presetVersion}`, {
+  name,
+  nxCloud: false,
+  packageManager: 'npm',
+});
+```
+
+This will invoke the `my-plugin` package's `preset` generator, which will contain the logic for setting up the workspace. This preset generator will be invoked when running either `npx create-nx-workspace --preset my-plugin` or `npx create-my-plugin`. For more information about customizing your preset, see: [Creating a Preset](/extending-nx/recipes/create-preset).
+
+## Testing your create package
+
+Because your create-my-plugin package will install your plugin package at runtime, both packages must be published in order to run them and see the results. To test your packages without making them publicly available, a `local-registry` target should be present on project in your workspace's root.
```

Comment:
```suggestion
Because your `create-my-plugin` package will install your plugin package at runtime, both packages must be published in order to run them and see the results. To test your packages without making them publicly available, a `local-registry` target should be present on the project in your workspace's root.
```